src/background/get-persisted-data.ts

```diff
@@ -9,21 +9,19 @@ export interface PersistedData {
     assessmentStoreData: AssessmentStoreData;
     userConfigurationData: UserConfigurationStoreData;
 }
-export function getPersistedData(indexedDBInstance: IndexedDBAPI): Promise<PersistedData> {
-    const persistedData = {} as PersistedData;
 
-    const promises: Array<Promise<any>> = [];
+const keyToPersistedDataMapping = {
+    [IndexedDBDataKeys.assessmentStore]: 'assessmentStoreData',
+    [IndexedDBDataKeys.userConfiguration]: 'userConfigurationData',
+};
+export function getPersistedData(indexedDBInstance: IndexedDBAPI, dataKeysToFetch: string[]): Promise<PersistedData> {
+    const persistedData = {} as PersistedData;
 
-    promises.push(
-        indexedDBInstance.getItem(IndexedDBDataKeys.assessmentStore).then(assessmentData => {
-            persistedData.assessmentStoreData = assessmentData;
-        }),
-    );
-    promises.push(
-        indexedDBInstance.getItem(IndexedDBDataKeys.userConfiguration).then(userConfig => {
-            persistedData.userConfigurationData = userConfig;
-        }),
-    );
+    const promises: Array<Promise<any>> = dataKeysToFetch.map(key => {
+        return indexedDBInstance.getItem(key).then(assessmentData => {
```

Comment:
`then(assessmentData)...` the name should be updated, right?

Reply:
Updated. Good catch.